pkg/task/task.go

```diff
@@ -27,24 +27,25 @@ type Task interface {
 
 // Command context maintains the mutable and shared entities.
 type CommandContext struct {
-	Bootstrapper       interfaces.Bootstrapper
-	Provider           providers.Provider
-	ClusterManager     interfaces.ClusterManager
-	GitOpsManager      interfaces.GitOpsManager
-	Validations        interfaces.Validator
-	Writer             filewriter.FileWriter
-	EksdInstaller      interfaces.EksdInstaller
-	PackageInstaller   interfaces.PackageInstaller
-	EksdUpgrader       interfaces.EksdUpgrader
-	CAPIManager        interfaces.CAPIManager
-	ClusterSpec        *cluster.Spec
-	CurrentClusterSpec *cluster.Spec
-	UpgradeChangeDiff  *types.ChangeDiff
-	BootstrapCluster   *types.Cluster
-	ManagementCluster  *types.Cluster
-	WorkloadCluster    *types.Cluster
-	Profiler           *Profiler
-	OriginalError      error
+	Bootstrapper           interfaces.Bootstrapper
+	Provider               providers.Provider
+	ClusterManager         interfaces.ClusterManager
+	GitOpsManager          interfaces.GitOpsManager
+	Validations            interfaces.Validator
+	Writer                 filewriter.FileWriter
+	EksdInstaller          interfaces.EksdInstaller
+	PackageInstaller       interfaces.PackageInstaller
+	EksdUpgrader           interfaces.EksdUpgrader
+	CAPIManager            interfaces.CAPIManager
+	ClusterSpec            *cluster.Spec
+	CurrentClusterSpec     *cluster.Spec
+	UpgradeChangeDiff      *types.ChangeDiff
+	BootstrapCluster       *types.Cluster
+	ManagementCluster      *types.Cluster
+	WorkloadCluster        *types.Cluster
+	Profiler               *Profiler
+	OriginalError          error
+	ManagementClusterState string
```

Comment:
I think now that it is part of the command context, we can be more clear with what the string represents.
Either `ManagementClusterStateDir` or `ManagementClusterStateFile`